README.md

```diff
@@ -1,234 +1,679 @@
+    Columns that should have no nulls but do: subject_id.
+    Columns that should have some non-nulls but don't: time
+
+```
+
+Alignment performs some modest type coercion and column re-ordering to return a version of the table that does
+conform to the schema, where possible (or it throws an error):
+
+```python
+>>> query_tbl = pa.Table.from_pydict({
+...     "time": [
+...         datetime(2021, 3, 1),
+...         datetime(2021, 4, 1),
+...         datetime(2021, 5, 1),
+...     ],
+...     "subject_id": [1.0, 2.0, 3.0],
+...     "code": ["A", "B", "C"],
+... })
+>>> Data.align(query_tbl)
+pyarrow.Table
+subject_id: int64
+time: timestamp[us]
+code: string
+----
+subject_id: [[1,2,3]]
+time: [[2021-03-01 00:00:00.000000,2021-04-01 00:00:00.000000,2021-05-01 00:00:00.000000]]
+code: [["A","B","C"]]
+
+```
+
+> [!NOTE]
+> See the [`flexible_schema`](https://flexible-schema.readthedocs.io/en/latest/) documentation for more
+> details on how the schema objects function.
+
+#### Canonical Codes
+
+The `code` column in the `Data` schema is a string that describes the type of measurement being observed. In
+general, there are _no restrictions_ on the vocabularies used in this column. However, MEDS does define two
+codes for general use that are likely to be applicable to nearly all MEDS datasets:
+
+| **Code**     | **Import**        | **Description**      | **Absolute or Prefix?**                                         |
+| ------------ | ----------------- | -------------------- | --------------------------------------------------------------- |
+| `MEDS_BIRTH` | `meds.birth_code` | The subject is born. | May be used either alone or as a prefix (e.g., `MEDS_BIRTH//*`) |
+| `MEDS_DEATH` | `meds.death_code` | The subject dies.    | May be used either alone or as a prefix (e.g., `MEDS_DEATH//*`) |
+
+```python
+>>> from meds import birth_code, death_code
+>>> print(f"{birth_code}, {death_code}")
+MEDS_BIRTH, MEDS_DEATH
+
+```
+
+### The `DatasetMetadata` schema
+
+The `DatasetMetadata` JSON schema structures essential information about the source dataset and its conversion
+to MEDS. It includes details such as the dataset’s name, version, and licensing, as well as specifics about
+the ETL process used for transformation. All fields in this schema are optional.
+
+Fields:
+
+1. `dataset_name`: The name of the dataset.
+2. `dataset_version`: The version of the dataset.
+3. `etl_name`: The name of the ETL process that generated the MEDS dataset.
+4. `etl_version`: The version of the ETL process.
+5. `meds_version`: The version of the MEDS format used.
+6. `created_at`: The creation date and time (in ISO 8601 format).
+7. `license`: The license under which the dataset is released.
+8. `location_uri`: The URI where the dataset is hosted.
+9. `description_uri`: The URI containing a detailed description of the dataset.
+10. `raw_source_id_columns`: A list of columns in the data that are not part of the core MEDS data schema
```

Comment:
I don't fully understand the content of these fields, but some of these names seem like they could be more concise. Could the detail be captured/accessed directly on the `Data` object instead? e.g.

```
>>> from flexible_schema import PyArrowSchema, Optional, Required
>>> import pyarrow as pa
>>> class Data(PyArrowSchema):
...     subject_id: Required(pa.int64(), nullable=False)
...     time: pa.timestamp("us")
...     code: Required(pa.string(), nullable=False)
...     numeric_value: Optional(pa.float32())
...     text_value: Optional(pa.string())
...     image_path: Optional(pa.string(), meds_role="code_modifier")
```